### PR TITLE
*: bunch of fixes...

### PR DESCRIPTION
**Pull request by runcom**
@mrunalp PTAL :)

### CODE REVIEW

**Comment by mrunalp** [comment]
Look fine. will wait for tests.

**Comment by runcom** [comment]
Fixing Travis failure...

**Comment by mrunalp** [comment]
@cevich https://atomic-e2e-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/cri-o-integration-tests-prs/1472/parameters/ never reported here :/ Could you please look into fixing this?

**Comment by runcom** [comment]
> @cevich https://atomic-e2e-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/cri-o-integration-tests-prs/1472/parameters/ never reported here :/ Could you please look into fixing this?

@cevich it seems something's wrong, tests pass but at the end we get:
```
20:35:36 cat: temp.txt: No such file or directory
20:35:36 grep: PLATFORM/*/job.properties: No such file or directory
20:35:36 grep: PLATFORM/*/job.properties: No such file or directory
```

**Comment by runcom** [comment]
:tada:

File: server/container_create.go [hunk 1]
```diff
@@ -124,29 +124,111 @@ func addImageVolumes(rootfs string, s *Server, containerInfo *storage.ContainerI
 	return nil
 }
 
+// resolveSymbolicLink resolves a possbile symlink path. If the path is a symlink, returns resolved
+// path; if not, returns the original path.
+func resolveSymbolicLink(path string) (string, error) {
```
Comment (rhatdan) [inline server/container_create.go]: fileutils.ReadSymlinkedDirectory(path) does something like this also.

**Comment by rhatdan** [comment]
LGTM

File: server/container_create.go (hunk 1 above)
Comment (runcom) [reply server/container_create.go]: Do you want me to use that? 
```
244 // ReadSymlinkedDirectory returns the target directory of a symlink.
245 // The target of the symbolic link may not be a file.
246 func ReadSymlinkedDirectory(path string) (string, error) {
247     var realPath string
248     var err error
249     if realPath, err = filepath.Abs(path); err != nil {
250         return "", fmt.Errorf("unable to get absolute path for %s: %s", path, err)
251     }
252     if realPath, err = filepath.EvalSymlinks(realPath); err != nil {                                                       253         return "", fmt.Errorf("failed to canonicalise path for %s: %s", path, err)
254     }
255     realPathInfo, err := os.Stat(realPath)
256     if err != nil {
257         return "", fmt.Errorf("failed to stat target '%s' of '%s': %s", realPath, path, err)
258     }
259     if !realPathInfo.Mode().IsDir() {
260         return "", fmt.Errorf("canonical path points to a file '%s'", realPath)
261     }
262     return realPath, nil
263 }
```

**Comment by mrunalp** [comment]
LGTM

**Comment by cevich** [comment]
> @cevich it seems something's wrong, tests pass but at the end we get:

Ewww, that can't be good.  I'll check it out...

**Pull request merged**